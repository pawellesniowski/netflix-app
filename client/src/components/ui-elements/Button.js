import React from 'react';

const Button = ({message}) => <button>{message}</button>;

export default Button;
