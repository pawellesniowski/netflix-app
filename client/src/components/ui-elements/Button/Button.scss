@import "../../../common/styles/colors.scss";

.button {
  width: 120px;
  height: 40px;
  margin: 4px;
  border-radius: 4px;
  border: 1px solid $color-netflix-red;
  text-decoration: none;
  font-size: 2rem;
  outline: $color-netflix-red;

  &:hover {
    box-shadow: 0 0 10px $color-netflix-red;
  }

  &:focus {
    box-shadow: none;
  }
}

.option-button {
  background-color: $color-netflix-red;
}

.submit-button {
  background-color: $color-netflix-red;
}
