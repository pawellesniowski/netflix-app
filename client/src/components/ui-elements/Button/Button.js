import React from "react";
import "./Button.scss";

const Button = ({ message, mode = "default" }) => (
  <button
    className={`button ${
      mode == "default" ? "option-button" : "submit-button"
    }`}
  >
    {message}
  </button>
);

export default Button;
