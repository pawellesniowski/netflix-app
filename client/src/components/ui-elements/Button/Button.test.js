//expect(element.html()).toEqual('<button class="btn">Search</button>');

import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button Component', () => {
  test("Snapshot test", () => {
    const button = shallow(<Button />);
    expect(button).toMatchSnapshot();
  });

  it("Should display 'submit' message by default", () => {
    const button = shallow(<Button />);
    expect(button.html()).toEqual('<button class="button option-button">Submit</button>');
  });

  it("Should have label like given in props", () => {
    const button = shallow(<Button message="go"/>);
    expect(button.html()).toEqual('<button class="button option-button">go</button>');
  });
});
