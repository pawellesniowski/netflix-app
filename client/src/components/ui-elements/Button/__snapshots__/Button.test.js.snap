// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot test 1`] = `ShallowWrapper {}`;
